### DISCUSSION TITLE
Uptime monitoring

### DISCUSSION THREAD

**Discussion by bohdan-s** [alexjustesen/speedtest-tracker#1367] [Ideas]
Is there any appetite to add in uptime monitoring?
Would be great to have an availability / uptime / outages data. 

Could be as simple as a ping to 8.8.8.8 or 1.1.1.1 every 30secs or so.

**Comment by alexjustesen** [comment]
You talking Internet uptime monitoring or website?

For websites I'd check out UptimeKuma.

**Reply by OnoBeachBum** [reply]
Along similar lines, I had an hour-long internet outage today, and it was recorded in the results, but the data was skipped in the plots. I would like the option to show the gaps in the plots to quickly visualize the outage. Is this possible?

**Comment by bohdan-s** [comment]
Internet uptime. 
We are already testing the internet speed on schedule, so this would just be a connectivity test. 

There is a commercial implementation of this that SamKnows provides to the ACCC for broadband monitoring in Australia. It does a regular speed test as well as a connectivity test and then provides a dashboard to see that data. 

Some of us are stuck on coax and xDSL technologies in Australia to uptime of the connection is as important as the speed.

**Reply by alexjustesen** [reply]
Makes perfect sense, a lot of Internet here in the States isn't much better. I'm a lucky small percentage that has fiber. 

I think you'll be able to achieve this with the upcoming ping tests. I can't say when it'll be done but it's already something I've started playing with locally.

**Reply by svenvg93** [reply]
Let me know if you need help with it. Happy to see where I can help and contribute

**Reply by alexjustesen** [reply]
It'll be after `v0.20.0`, I want to upgrade to PHP and Laravel and then shut down my image before adding more features. Need to prioritize some spring cleaning to reduce overhead.

**Reply by svenvg93** [reply]
Sounds good! Let me know if and when I can help with anything. Also happy to try and fix some of the reported issues 😊
